dask/bag/core.py

```diff
@@ -292,7 +292,9 @@ def from_delayed(value):
 
         See ``dask.bag.from_delayed`` for details
         """
-        from dask.delayed import Delayed
+        from dask.delayed import Delayed, delayed
+        if not isinstance(value, Delayed) and hasattr(value, 'key'):
+            value = delayed(value)
```

Comment:
Why the check for `'key'` here but not elsewhere (I assume this is to duck-type on futures)?